### PR TITLE
Add climate support

### LINKED ISSUE #93: Thermostat/Climate support
Hey guys, thank you all for the nice work you did.
Would it be possible to add thermostat/climate support?

### PR DESCRIPTION
**Pull request by robbiet480**
Closes #93 for real

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @tfresh1, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;homebridge-homeassistant#104).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by homeassistant** [comment]
Hi @tfresh1, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;homebridge-homeassistant#104).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by homeassistant** [comment]
Hi @tfresh1, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;homebridge-homeassistant#104).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by homeassistant** [comment]
Hi @tfresh1, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;homebridge-homeassistant#104).

Once you do that we will be able to review and accept this pull request.

Thanks!

**Comment by jschwalbe** [comment]
I'm trying to figure out how to test this.. On the other issue (#93?) there was a npm command which I ran and couldn't get it to function correctly. Any hints? Thanks

**Comment by luxus** [comment]
I just tested this branch and my thermostat just show the temperature. No option to change anything

**Comment by thechrisreese** [comment]
I have been working on something similar offline. This would be awesome. Between this and adding Kēvo to Home Assistant, it would be a dream. I'm going to try this out.

**Comment by Minims** [comment]
The thermostat is not loaded if generic_thermostat. 
The accessory seems not build. `accessory` is null when `entity_type` is `climate`.
I think something is missing for this virtual thermostat, but I've still not found exactly what in my debug.

**Comment by ssolmaz** [comment]
I am waiting for the day to support the generic thermostat.
I wish you success in your work.

**Comment by Minims** [comment]
I've made an alpha version on my side, it's working, but it seems the only thing you can do with the generic is to set up the temperature.  operation_mode or set_aux_heat make no change in hass when I make the API call. I will put my version in pull request during the week.

**Comment by thechrisreese** [comment]
I have tried this with my Wink setup but it is nowhere near correct. It pulls the mode but can't find thresholds, or targets

**Comment by robbiet480** [comment]
Hey all, I don't have any climate devices to test with, someone else with a climate device is going to have to get your issues fixed. :(

**Comment by Minims** [comment]
I've pushed my branch of alpha version of generic_thermostat if someone wants to test ;-)

**Comment by Shigawire** [comment]
@robbiet480 would you be so kind to merge the current master into this branch?

I would like to test the thermostate feature (got the Danfoss LC-13 at home) but there are syntax errors with this brach (specifically a feature that has been fixed in the current master). 
Cheers

**Comment by robbiet480** [comment]
@Shigawire done

File: index.js [hunk 1]
```diff
@@ -175,6 +176,8 @@ HomeAssistantPlatform.prototype = {
           accessory = HomeAssistantSensorFactory(that.log, entity, that);
         } else if (entityType === 'device_tracker') {
           accessory = HomeAssistantDeviceTrackerFactory(that.log, entity, that);
+        } else if (entityType === 'climate') {
+          accessory = HomeAssistantClimate(that.log, entity, that);
```
Comment (Shigawire) [inline index.js]: this should be ```new HomeAssistantClimate(that.log, entity, that);```

**Comment by Shigawire** [comment]
@robbiet480 oh yeah, also adding ```climate``` as available service to the homebridge's config.json shouldn't be missing. Took me a while to figure out why it didn't work until I realized that I did not enable climate in the homebridge's config.

**Comment by tfresh1** [comment]
Hey guys, sorry for being away such a long time. 
I just updated to the optimized and clean code and it works perfect for my homematic thermostat.
The only thing you should have to do is add the climate to the available service list in the config and  add the 'new' in the index.js like @Shigawire said.

**Comment by robbiet480** [comment]
@Minims all merged. Feel free to make changes on a branch. Will wait a couple of days to get your stuff merged in and then will release to npm.

**Comment by javefang** [comment]
Hi @robbiet480 , any plan to make a new release?

**Pull request merged**